src/Microsoft.IdentityModel.Tokens/EventBasedLRUCache.cs

```diff
@@ -30,61 +29,59 @@ namespace Microsoft.IdentityModel.Tokens
     internal class EventBasedLRUCache<TKey, TValue>
     {
         internal delegate void ItemRemoved(TValue Value);
+        internal delegate bool ShouldRemove(TValue Value);
 
         private readonly int _capacity;
-
         // The percentage of the cache to be removed when _maxCapacityPercentage is reached.
```

Comment:
This comment belongs with the `_compactionPercentage` field (which could be made a `const`).